pkg/auto/forward.port.go

```diff
@@ -0,0 +1,161 @@
+		git:                     ld.Git,
+		VR:                      ld.VR,
+		assetsToBeForwardPorted: assetsToPort,
+		pullRequests:            make(map[string]PullRequest),
+		forkRemoteURL:           forkURL,
+	}, err
+}
+
+// ExecuteForwardPort will execute all steps to organize and create the forward-port PRs
+func (fp *ForwardPort) ExecuteForwardPort(chart string) error {
+	// Get the forward port script commands
+	commands, err := fp.createForwardPortCommands(chart)
+	if err != nil {
+		return err
+	}
+	// Organize the commands into pull requests grouping by chart with it's dependencies
+	fp.organizePullRequestsByChart(commands)
+	// Execute the forward port commands
+	return fp.executeForwardPorts()
+}
+
+// organizePullRequests will organize the commands into pull requests
+func (fp *ForwardPort) organizePullRequestsByChart(commands []Command) {
+	lastChart := ""
+
+	for _, command := range commands {
+
+		changed := checkIfChartChanged(lastChart, command.Chart)
+
+		// If the chart is the same as the last chart, append the command to the pull request
+		if !changed {
+			pr := fp.pullRequests[lastChart]           // Extract the struct from the map
+			pr.commands = append(pr.commands, command) // Modify the struct's field
+			fp.pullRequests[lastChart] = pr            // Put the modified struct back into the map
+		}
+
+		// If the chart is different from the last chart, organize a new pull request
+		if changed {
+			pr := PullRequest{
+				branch:   fmt.Sprintf("auto-forward-port-%s-%.1f", command.Chart, fp.VR.BranchVersion),
+				commands: []Command{command},
+			}
+			fp.pullRequests[command.Chart] = pr
+			lastChart = command.Chart
+		}
+	}
+}
+
+// executeForwardPorts will execute the forward-port commands.
+// It will create a new branch to forward-port the assets, clean the release.yaml file and commit.
+// After each forward-port execution it will add and commit the changes to the git repository.
+// It will push the branch to the remote repository and delete the local branch before moving,
+// to the next Pull Request.
+func (fp *ForwardPort) executeForwardPorts() error {
+	// save the original branch to change back after forward-port
+	originalBranch := fp.git.Branch
+
+	// create log file
+	fpLogs, err := lifecycle.CreateLogs("forward-ported", "")
+	defer fpLogs.File.Close()
+	if err != nil {
+		return err
+	}
+
+	// write log title and header
+	fpLogs.WriteHEAD(fp.VR, "Forward-Ported Assets")
+
+	for asset, pr := range fp.pullRequests {
+		fpLogs.Write(asset, "INFO")
+
+		// open and check if it is clean the git repo
+		err := fp.createNewBranchToForwardPort(pr.branch)
+		if err != nil {
+			return err
+		}
+
+		// clean release.yaml in the new branch
+		if err = prepareReleaseYaml(); err != nil {
+			return err
+		}
+		// git add && commit cleaned release.yaml
+		if err = fp.git.AddAndCommit("cleaning release.yaml"); err != nil {
+			return err
+		}
+
+		fpLogs.Write(fmt.Sprintf("Branch: %s", pr.branch), "INFO")
```

Comment:
```suggestion
		fpLogs.Write("Branch: " + pr.branch, "INFO")
```

When concatenating strings, there's no need to use any of the XprintfX functions. They're incredibly heavy due to the amount of reflection involved.

Reply:
Fixing